src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/SolutionExplorerInProcess.cs

```diff
@@ -553,21 +554,47 @@ private async Task<string> WaitForBuildToFinishAsync(IVsOutputWindowPane buildOu
         {
             await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
-            await KnownUIContexts.SolutionExistsAndNotBuildingAndNotDebuggingContext;
```

Comment:
So why didn't this work in the first place?

Reply:
Hmm, we execute the BuildSln command before entering this method. I can only imagine that we are reaching here prior to the build actually beginning.

Reply:
that seems broken and indicates to me some other component not properly updating the UI context in response to that.

Reply:
This reminds me of the solution/project save async issue that popped up recently. They made the file save async which broke us. They then updated so that it was only async when invoked through the UI, not when the command was executed through code.

Reply:
See https://github.com/dotnet/roslyn/pull/62117 for that issue

Reply:
> I can only imagine that we are reaching here prior to the build actually beginning.

Does this now create the reverse problem? If the build happens so quickly that we don't actually get here until it's already done, and we will hang forever? I'd expect any subscription needs to be created before we issue the build command.

Reply:
Good point. Updated.